lib/actions/add.js

```diff
@@ -17,52 +17,31 @@ const Boom = require('boom');
 
 module.exports = function addToCollection(route, options) {
 
-    return (request, reply) => {
+    return async (request, h) => {
 
         const actionUtil = require('../actionUtil')(request, options);
 
         // Ensure a model and alias can be deduced from the request.
         const Model = actionUtil.modelFromParam(options.model);
 
         const keys = actionUtil.getKeys();
+        const foundModel = await Model.query().findById(keys.parent.value);
 
-        Model.query().findById(keys.parent.value).asCallback((modelErr, foundModel) => {
+        if (!foundModel) {
+            return Boom.notFound('No record found with the specified `id`.');
```

Comment:
If you're feeling particularly offended you can always `throw` this error too, and it has the same effect :)